generate.sh

```diff
@@ -65,13 +65,88 @@ import (
 
 "
 
-# read_enum read C enum values.
+# read_enum reads C enum values.
 # arg1 - enum name in C code.
 # arg2 - path to a C code.
 function read_enum {
 	grep -Pzo "(?s)enum ${1} \{.+?\};" "${2}" | grep -aP "^\t"
 }
 
+# read_define_prefix_name reads C name of the prefix for constants.
+# arg1 - name of the define with constants.
+# arg2 - enum name in C code.
+# arg3 - path to a C code.
+function read_define_prefix_name {
+	read_enum $2 $3 | grep -oP "${1}\(.+?\)" | grep -oP "\(.+?\)" | \
+		# Delete first and last symbols from the string ("(" and ")" respectively).
+		sed "s/^.//;s/.$//"
+}
+
+# read_define reads C constants from define.
+# arg1 - name of the define with constants.
+# arg2 - path to a C code.
+function read_define {
+	sed -n "/#define ${1}/,/^$/p" "${2}"
+}
+
+# read_prefix reads prefix for constants from define.
+# arg1 - name of the define, containing the prefix.
+# arg2 - path to a C code.
+function read_prefix {
+	read_define $1 $2 | grep -oP "\) .+?#" | \
```

Comment:
nit: All this can be done with one call to awk

Reply:
Updated (last sed from this function now merged with awk, but I left the grep to use only the matched strings, not the whole lines).